### PR TITLE
spectator date time

### PR DESCRIPTION
**Pull request by shivangi-gupta123**

### CODE REVIEW

File: src/app/spectatorDashboard/Form/page.tsx [hunk 1]
```diff
@@ -118,12 +118,131 @@ const Form = ({ ...props }) => {
...
+        date: dateValue
+      });
+
+    }
+
+  }, [roomIdToUpdate])
+
+
+
+  const handleChange = (e: any) => {
```
Comment (Ashok-tgc) [inline src/app/spectatorDashboard/Form/page.tsx]: refactor this function

File: src/app/spectatorDashboard/Form/page.tsx [hunk 2]
```diff
@@ -118,12 +118,131 @@ const Form = ({ ...props }) => {
...
+    if (newImages.length === 0) {
+      setFieldValue('mapImg', null);
+    }
+  };
+
+  ;
+
   return (
     <>
       <div>
```
Comment (Ashok-tgc) [inline src/app/spectatorDashboard/Form/page.tsx:239]: split into components

File: src/app/spectatorDashboard/Room/page.tsx [hunk 3]
```diff
@@ -60,7 +61,6 @@ const Room = () => {
       const spectatorResponse = await sendRequest('room/user-rooms', {
```
Comment (Ashok-tgc) [inline src/app/spectatorDashboard/Room/page.tsx:65]: create a service for this

File: src/app/spectatorDashboard/Form/page.tsx (hunk 1 above)
Comment (shivangi-gupta123) [reply src/app/spectatorDashboard/Form/page.tsx]: done

**Pull request merged**